File: Ontology/IDataCardinality.cs
namespace Ontology
{
    public interface IDataCardinality: IDataPropertyRestriction
    {
        int        Cardinality { get; }
        IDataRange DataRange   { get; }
    }

    public interface IDataMinCardinality: IDataCardinality
    {
    }

    public interface IDataMaxCardinality: IDataCardinality
    {
    }

    public interface IDataExactCardinality: IDataCardinality
    {
    }
}

File: Web/ClientApp/src/app/Ontology/IDataCardinality.ts
import { ICardinality } from "./ICardinality";
import { IDataPropertyRestriction } from "./IDataPropertyRestriction";
import { IDataRange } from "./IDataRange";

export interface IDataCardinality extends ICardinality, IDataPropertyRestriction
{
    readonly DataRange: IDataRange;
}

export interface IDataMinCardinality extends IDataCardinality
{
}

export interface IDataMaxCardinality extends IDataCardinality
{
}

export interface IDataExactCardinality extends IDataCardinality
{
}
